06-files-input-and-output/high_scores.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2016 Ami . All rights reserved
+
+import csv
+import os
+
+def parse_csv(file):
+    data = {}
+    print
+    with open(file, "r") as csv_file:
+        try:
+            csv_reader = csv.reader(csv_file)
+            for name, score in csv_reader:
+                score = int(score)
+                if name not in data:
+                    data[name] = score
+                else:
+                    if score > data[name]:
+                        data[name] = score
+
+        except ValueError:
+            print("Could not parse csv")
+
+    return data
+
+def show_high_scores(data):
+    print("High Scores:")
+    print("-------------")
+
+    for name in sorted(data):
+        print(name, data[name])
+
+def leader_board(data):
+    print("Leader board:")
+    print("-------------")
+    sorted_data = sorted(data, key=data.get, reverse=True)
+
+    for name in sorted_data:
```

Comment:
Instead of looking for each element's index inside the sorted array which is inefficient (do you understand why?) use the `enumerate` function (https://docs.python.org/2/library/functions.html#enumerate).